Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -2109,10 +2109,8 @@ void FrameLoader::stopForBackForwardCache()
     if (RefPtr documentLoader = m_documentLoader)
         documentLoader->stopLoading();
 
-    for (RefPtr child = m_frame->tree().firstChild(); child; child = child->tree().nextSibling()) {
-        if (RefPtr localChild = dynamicDowncast<LocalFrame>(child.get()))
-            localChild->protectedLoader()->stopForBackForwardCache();
-    }
+    for (RefPtr child = m_frame->tree().firstChild(); child; child = child->tree().nextSibling())
+        child->stopForBackForwardCache();
```

Comment:
So is it a goal to remove use of Local/RemoteFrame dynamicDowncasts in preference for Frame methods that self-dispatch? (essentially the visitor pattern)

Reply:
I think it depends on what the function does. For recursive functions like this one, having a function on Frame that skips over each RemoteFrame makes sense, but there may be other scenarios where it isn’t necessary.